modules/reindex/src/main/java/org/elasticsearch/index/reindex/AbstractAsyncBulkByScrollAction.java

```diff
@@ -116,10 +118,18 @@
 
     public AbstractAsyncBulkByScrollAction(BulkByScrollTask task, boolean needsSourceDocumentVersions,
```

Comment:
I would prefer to remove this constructor, and  just keep the one with the  two extra parameters. Makes it a more conscious choice for callers  to pass `null` as parameter